packages/next/build/analysis/get-page-static-info.ts

```diff
@@ -10,9 +11,17 @@ import { promises as fs } from 'fs'
 import { tryToParsePath } from '../../lib/try-to-parse-path'
 import * as Log from '../output/log'
 import { SERVER_RUNTIME } from '../../lib/constants'
+import { checkCustomRoutes } from '../../lib/load-custom-routes'
 
-interface MiddlewareConfig {
-  pathMatcher: RegExp
+export interface MiddlewareConfig {
+  matchers: MiddlewareMatcher[]
+}
+
+export interface MiddlewareMatcher {
+  regexp: string
+  basePath?: false
+  locale?: false
+  has?: RouteHas[]
 }
```

Comment:
I want to put this type to somewhere else since it's referenced from many locations but not sure where it should be. Maybe somewhere under `shared/lib`?

Reply:
If we want this publicly available we can export it from `next/types/index.d.ts` if it's meant to only be internal keeping it here or a shared lib sounds good.

Reply:
It's only for internal use so will keep as it is for now